### DISCUSSION TITLE
Do I need to reference `Microsoft.Maui.Controls.Compatibility` in my purely MAUI project (.NET 8)? Why?

### DISCUSSION THREAD

**Discussion by meinmunzen** [dotnet/maui#21501] [Q&A] [answered]
Official template [contains](https://github.com/dotnet/maui/blob/c7d1a4ec8d857aba674362d2777d98855f0ca67a/src/Templates/src/templates/maui-mobile/MauiApp.1.csproj#L60-L61) both `Microsoft.Maui.Controls` and `Microsoft.Maui.Controls.Compatibility`. 

However, it's not really clear to me why a new .NET project requires compatibility functionality. Especially, for cases when one does not migrate from Xamarin Forms to MAUI.

Could anyone explain it a bit more?

**Comment by davidortinau** [comment, marked as answer]
Right now if you remove it, it is still going to be referenced when needed by the `UseMaui` project capability (like pre-net8). The PackageReference is there to allow setting the version to get a NuGet instead of the workload installed version.

As we get through the late wave of Xamarin upgrades, we will completely detangle and remove it from templates. Likely a .NET 9 task.

**Reply by bcaceiro** [reply]
Since we are now in net8 , can we safely remove it from dependencies? Or will it still be referenced?

**Reply by FaithfulDev** [reply]
@bcaceiro Just in case you missed it, in net9 they removed it from the template:
https://github.com/dotnet/maui/pull/22203
